client/my-sites/upgrades/cart/cart-item.jsx

```diff
@@ -113,17 +159,27 @@ module.exports = React.createClass( {
 					volume: cartItem.volume,
 					productName: cartItem.product_name
 				}
-			};
-
+			},
+			hasDomainSuffix, gappsProductName;
+		if ( isDomainRegistration( cartItem ) ) {
+			return cartItem.product_name;
+		}
 		if ( ! cartItem.volume ) {
 			return cartItem.product_name;
 		} else if ( cartItem.volume === 1 ) {
 			switch ( cartItem.product_slug ) {
 				case 'gapps':
+					hasDomainSuffix = cartItem.product_name.indexOf( cartItem.meta ) > -1;
```

Comment:
`includes` might be really helpful here (`String.prototype.includes()` as of ES6) if these are string values, **lodash** has a more generic `includes()` if not

```js
hasDomainSuffix = cartItem.product_name.includes( cartItem.meta );
```